src/main/java/duke/tasks/TaskList.java

```diff
@@ -0,0 +1,200 @@
+package duke.tasks;
+
+import duke.storage.Storage;
+import duke.exception.DukeException;
+import duke.ui.Ui;
+
+import java.lang.reflect.Array;
+import java.util.ArrayList;
+
+/**
+ * Represents a Task List to store User's Tasks.
+ */
+public class TaskList {
+    /** The Indentation Level to format text. */
+    private static final String INDENTATION = Ui.INDENTATION;
+
+    /** The ArrayList to represent the Task List. */
+    private ArrayList<Task> taskList;
+
+    /** The Storage for the Chatbot. */
+    private Storage storage;
+
+
+    /**
+     * The Constructor for the TaskList.
+     *
+     * @param taskList The ArrayList to initialise as the TaskList.
+     * @param storage The Storage of the Chatbot.
+     */
+    public TaskList(ArrayList<Task> taskList, Storage storage) {
+        this.taskList = taskList;
+        this.storage = storage;
+    }
+
+    /**
+     * Formats the Task List to output.
+     *
+     * @return The formatted task list as a String.
+     */
+    public String formatList() {
+        String str = "";
+        int len = taskList.size();
+        for (int i = 0; i < len; i++) {
+            str = str + (i + 1) + ". " + taskList.get(i);
+            if (i != (len - 1)) {
+                str += "\n " + INDENTATION;
+            }
+        }
+        return str;
+    }
+
+    /**
+     * Handles the Marking of the Tasks in the Task List.
+     *
+     * @param commandNum The Index of the Task as a String.
+     * @param status The status of the marking as either 'mark' or 'unmark'.
+     * @return The Success output as a String.
+     * @throws DukeException If Task Index provided is invalid.
+     */
+    public String handleMarking(String commandNum, String status) throws DukeException {
+
+        try {
+            int index = Integer.parseInt(commandNum) - 1;
+            String buffer = "";
+
+            if (index > taskList.size() - 1 || index < 0) {
+                throw new DukeException("Invalid duke.tasks.Task Index provided!");
+            }
+            Task selectedTask = taskList.get(index);
+            if (status.equals("mark")) {
+                selectedTask.markTask();
+                buffer = "Nice! I've marked the task as done:\n   " +
+                        INDENTATION + selectedTask;
```

Comment:
Perhaps you could add the line break before the `+` operator.
```suggestion
                buffer = "Nice! I've marked the task as done:\n   "
                         + INDENTATION + selectedTask;
```